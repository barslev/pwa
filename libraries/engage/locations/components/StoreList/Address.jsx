import React from 'react';
import PropTypes from 'prop-types';
import { LocationIcon } from '@shopgate/engage/components';
import { i18n } from '@shopgate/engage/core';
import { address as container, addressIcon } from './style';

/**
 * Renders the a pickup location's address information.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function Address({ address }) {
  if (!address) {
    return null;
  }

  return (
    <div className={container}>
      <div className={addressIcon}>
        <LocationIcon />
      </div>
      <div>
        <div data-test-id="street">
          {address.street}
        </div>
        {(address.street2 && address.street2 !== '') && (
          <div data-test-id="street2">
            {address.street2}
          </div>
        )}
        {(address.street3 && address.street3 !== '') && (
          <div data-test-id="street3">
            {address.street3}
          </div>
        )}
        {(address.street4 && address.street4 !== '') && (
          <div data-test-id="street4">
            {address.street4}
          </div>
        )}
        {i18n.text('product.location.address', address)}
      </div>
    </div>
  );
}

Address.propTypes = {
  address: PropTypes.shape({
    city: PropTypes.string,
    postalCode: PropTypes.string,
    region: PropTypes.string,
    street: PropTypes.string,
    street2: PropTypes.string,
    street3: PropTypes.string,
    street4: PropTypes.string,
  }),
};

Address.defaultProps = {
  address: null,
};

export default Address;
