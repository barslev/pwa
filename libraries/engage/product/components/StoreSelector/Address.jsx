import React from 'react';
import PropTypes from 'prop-types';
import LocationIcon from '@shopgate/pwa-ui-shared/icons/LocationIcon';
import I18n from '@shopgate/pwa-common/components/I18n';
import { address as container, addressIcon } from './style';

/**
 * Renders the a pickup location's address information.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function Address({ address }) {
  if (!address) {
    return null;
  }

  const {
    street, street2, street3, street4,
  } = address;

  return (
    <div className={container}>
      <div className={addressIcon}>
        <LocationIcon />
      </div>
      <div>
        <div data-test-id="street">{street}</div>
        {street2 && street2 !== '' && <div data-test-id="street2">{street2}</div>}
        {street3 && street3 !== '' && <div data-test-id="street3">{street3}</div>}
        {street4 && street4 !== '' && <div data-test-id="street4">{street4}</div>}
        <I18n.Text string="product.location.address" params={address} />
      </div>
    </div>
  );
}

Address.propTypes = {
  address: PropTypes.shape({
    city: PropTypes.string,
    postalCode: PropTypes.string,
    region: PropTypes.string,
    street: PropTypes.string,
    street2: PropTypes.string,
    street3: PropTypes.string,
    street4: PropTypes.string,
  }),
};

Address.defaultProps = {
  address: null,
};

export default Address;
