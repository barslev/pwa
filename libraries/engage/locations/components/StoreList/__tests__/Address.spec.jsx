import React from 'react';
import { shallow } from 'enzyme';
import Address from '../Address';

const dummyAddress = {
  street: 'Test Street 123',
  city: 'Testcity',
  postalCode: '12345',
  region: 'TX',
  street3: '',
};

describe('<Address />', () => {
  it('should not render if no address is passed in', () => {
    const wrapper = shallow(<Address />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render as expected', () => {
    const wrapper = shallow(<Address address={dummyAddress} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('LocationIcon')).toBeTruthy();
    expect(wrapper.find('[data-test-id="street"]').text()).toEqual(dummyAddress.street);
    expect(wrapper.find('[data-test-id="street2"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="street3"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="street4"]').getElements().length).toEqual(0);
    expect(wrapper.find('Translate')).toBeTruthy();
  });

  it('should render more streets if necessary', () => {
    const wrapper = shallow(<Address
      address={{
        ...dummyAddress,
        street2: 'Some additional street info',
      }}
    />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('LocationIcon')).toBeTruthy();
    expect(wrapper.find('[data-test-id="street"]').text()).toEqual(dummyAddress.street);
    expect(wrapper.find('[data-test-id="street2"]').text()).toEqual('Some additional street info');
    expect(wrapper.find('[data-test-id="street3"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="street4"]').getElements().length).toEqual(0);
    expect(wrapper.find('Translate')).toBeTruthy();
  });
});
