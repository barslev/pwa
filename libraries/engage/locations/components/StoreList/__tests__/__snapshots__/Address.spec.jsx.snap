// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Address /> should not render if no address is passed in 1`] = `""`;

exports[`<Address /> should render as expected 1`] = `
<div
  className={
    Object {
      "data-css-1npkmgj": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-1rqdg8e": "",
      }
    }
  >
    <Location />
  </div>
  <div>
    <div
      data-test-id="street"
    >
      Test Street 123
    </div>
    product.location.address
  </div>
</div>
`;

exports[`<Address /> should render more streets if necessary 1`] = `
<div
  className={
    Object {
      "data-css-1npkmgj": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-1rqdg8e": "",
      }
    }
  >
    <Location />
  </div>
  <div>
    <div
      data-test-id="street"
    >
      Test Street 123
    </div>
    <div
      data-test-id="street2"
    >
      Some additional street info
    </div>
    product.location.address
  </div>
</div>
`;
